src/controllers/controller.polarArea.js

```diff
@@ -138,6 +138,32 @@ module.exports = function(Chart) {
 			});
 		},
 
+		getStartAngle: function(customAngles, dataSetData, metaData, datasetStartAngle, circumference, index) {
+
+			// If there is NaN data before us, we need to calculate the starting angle correctly.
+			// We could be way more efficient here, but its unlikely that the polar area chart will have a lot of data
+
+			var previousAngle = 0;
+			var visibleCount = 0;
+			var startAngle = 0;
+			var i = 0;
+
+			for (i = 0; i < index; ++i) {
```

Comment:
Do we really need to iterate every time for each element? can't we use angle `at index-1` to calculate the angle of the element at `index`

Reply:
I think we have to iterate since the array (or function/number) indicate the angle for one item and not it's starting angle. So to get the starting angle of the current element we need to calculate it from the start angle of the chart. Or would I have missed something?

Reply:
Yes, that's right.

Reply:
What I mean is that we could compute a `_starts` and `_angles` private cache arrays in `update()` that contains all start angles in which case only one iteration is needed, something like:

```javascript
update: function(...) {
    // ...
    var start = options.startAngle || 0;
    for (i = 0; i < count; ++i) {
        this._starts[i] = start;
        angle = [compute slice angle]; // if dataset invisible, angle == 0
        this._angles[i] = angle;
        start += angle;
    }
    // ...
    helpers.each(meta.data, function(arc, index) {
        me.updateElement(arc, index, reset);
    });
}

updateElement: function(...) {
   // ...
   var startAngle = this._starts[index];
   var endAngle = startAngle + this._angles[index];
   // ...
}
```

It seems to simplify a lot the computation / logic.

Reply:
I hadn't thought about that but yes it could save some performance.